build.proj

```diff
@@ -106,8 +122,12 @@
                                     NewVersion="$(NewVersion)" />
   </Target>
 
+  <PropertyGroup>
+    <RestorePackagesDependsOn>BatchRestorePackages</RestorePackagesDependsOn>
```

Comment:
I think you can just add a `BeforeTargets=BatchRestorePackages` on your GenerateAllTestProjectJsons to accomplish this scenario.